code/main/task/eventProcessor.brs

```diff
@@ -31,22 +31,57 @@ function _adb_EventProcessor(task as object) as object
 
             if _adb_isRequestEvent(event)
                 _adb_logInfo("handleEvent() - handle event: " + FormatJson(event))
-                if event.apiName = m._CONSTANTS.PUBLIC_API.SEND_EDGE_EVENT
-                    m._sendEvent(event)
-                    return
-                else if event.apiName = m._CONSTANTS.PUBLIC_API.SET_CONFIGURATION
-                    m._setConfiguration(event)
-                    return
-                else if event.apiName = m._CONSTANTS.PUBLIC_API.SET_LOG_LEVEL
-                    m._setLogLevel(event)
-                    return
-                else if event.apiName = m._CONSTANTS.PUBLIC_API.SET_EXPERIENCE_CLOUD_ID
-                    m._setECID(event)
-                    return
-                else if event.apiName = m._CONSTANTS.PUBLIC_API.RESET_IDENTITIES
-                    m._resetIdentities(event)
-                    return
+                if m._task <> invalid and m._task.hasField("debugInfo")
+                    loggingService = _adb_serviceProvider().loggingService
+
+                    networkService = _adb_serviceProvider().networkService
+                    networkService._debugMode = true
+                    ' -------------------
+                    m._processAdbRequestEvent(event)
```

Comment:
I don't think we should be adding one more function here, previous code was okay here.

Reply:
The previous code was starting to look messy after including collecting debug info code before and after the event processing block, so I extracted a new function here to make the code clearer.  As we discussed, we will add try-catch logic around event processing as well, it would be easier to add it around  `_processAdbRequestEvent` function.